enhancements/installer/kubevirt-ipi.md

```diff
@@ -0,0 +1,346 @@
+    before running the installer, and the created networkName (NAD) will be the input for the installer.  
+
+    
+- Storage
+
+    CSI driver for `KubeVirt` is not available yet.
+
+
+## Design Details
+
+- Namespaces
+
+    For each tenant cluster we will create a namespace with the ClusterID.
+
+    *Open question: should the namespace creation done by the user or by the installer.*
+
+- Images
+
+    - Option 1 - For each namespace we will create DataVolume (CDI CRD) with RHCOS image which will be cloned for
+    each VM, masters and workers.
+    - Option 2 - We will use URL and each VM will pull the image for itself without the need for cloning.
+ 
+- Network
+
+    ####Option 1 - Pods network
+    - set cluster baseDomain as svc.cluster.local so services that we will create
+    as LoadBalancer will have the expected FQDN <service-name>.<namespace>.svc.cluster.local
+    - Create VMs with one interface on pods network.
+    - Create headless services for each VM to create DNS records for internal communication
+    between the nodes.
+    - Create services for 'api' and 'api-int' as loadbalancers between the masters
+    with MCS port (26223) and API server port (6443)
+    - Set ingress domain name for default router as subdomain of the ingress domain
+    of the infra OCP.
+    - Create in the underlying OCP route for each route in the provisioned cluster with
+    hostname value as the route on the provisioned cluster.\
+    Alternatively, if the infra OCP support wildcard route then one route from
+    type subdomain can be defined for all the routes to the provisioned cluster.
+    - Isolation will be achieved by creation of network policies that will allow traffic
+    only between VMs(pods) that are related to the same provisioned cluster.
+    
+    ####Option 2 - Secondary network (Multus)
+    - Create VMs attached to the secondary network (NAD) that was configured.
+    - Isolation achieved by the secondary network, it's up to the admin to decide how to create the
+    secondary networks that can use different VLAN/VXLAN/etc. 
+    
+- Storage
+
+    The VMs boot volumes will be PVs allocated from the infra cluster
+    
+    For PVs requested for pods running on the tenant cluster we have a few options:
+
+    ####Option 1 - Direct storage CSI 
+   The provisioned cluster will use CSI to attach storage using network to the VM guests.
+   This can be OCS CSI driver to consume storage from OCS installed on the infra
+   OpenShift as a tenant of OCS or any other external storage.
+   
+   ####Option 2 - KubeVirt CSI driver
+   Develop CSI driver for KubeVirt platform.
+   
+   This provider should forward the request to the infra cluster to allocate PV
+   from the infra cluster storageClass and attach it to the relevant VM where the PV will be exposed to the guest
+   as block device that the driver will attach to the requested pods.
+
+- Anti-affinity
```

Comment:
We definitely need to do this, to prevent the cluster from being unrecoverable in the event of the loss of two master nodes.  We can use soft affinity, so things can come up in a demo environment.